src/mlpack/core/util/param.hpp

```diff
@@ -56,23 +53,117 @@ using DatasetInfo = DatasetMapper<IncrementPolicy, std::string>;
+ * present in your program!  Therefore, use it in the main.cpp
+ * (or corresponding executable) in your program.
+ *
+ * @see mlpack::IO, PARAM_FLAG(), PARAM_INT_IN(), PARAM_DOUBLE_IN(),
+ * PARAM_STRING_IN(), PARAM_VECTOR_IN(), PARAM_INT_OUT(), PARAM_DOUBLE_OUT(),
+ * PARAM_VECTOR_OUT(), PARAM_INT_IN_REQ(), PARAM_DOUBLE_IN_REQ(),
+ * PARAM_STRING_IN_REQ(), PARAM_VECTOR_IN_REQ(), PARAM_INT_OUT_REQ(),
+ * PARAM_DOUBLE_OUT_REQ(), PARAM_VECTOR_OUT_REQ(), PARAM_STRING_OUT_REQ().
+ *
+ * @param EXAMPLE Long string describing a simple usage example.. Newlines
+ *     should not be used here; this is taken care of by IO (however, you
+ *     can explicitly specify newlines to denote new paragraphs).  You can
+ *     also use printing macros like PRINT_CALL(), PRINT_DATASET(),
+ *     and others.
+ */
+#ifdef __COUNTER__
+  #define BINDING_EXAMPLE(EXAMPLE) static \
+      mlpack::util::Example \
+      JOIN(io_programexample_dummy_object_, __COUNTER__) = \
+      mlpack::util::Example( \
+      []() { return(std::string(EXAMPLE)); });
+#else
+  #define BINDING_EXAMPLE(EXAMPLE) static \
+      mlpack::util::Example \
+      JOIN(JOIN(io_programexample_dummy_object_, __LINE__), opt) = \
+      mlpack::util::Example( \
+      []() { return(std::string(EXAMPLE)); });
+#endif
+/**
+ * Document an executable.  Mutiple instance of this macro can be
+ * present in your program!  Therefore, use it in the main.cpp
+ * (or corresponding executable) in your program.
+ *
+ * @see mlpack::IO, PARAM_FLAG(), PARAM_INT_IN(), PARAM_DOUBLE_IN(),
+ * PARAM_STRING_IN(), PARAM_VECTOR_IN(), PARAM_INT_OUT(), PARAM_DOUBLE_OUT(),
+ * PARAM_VECTOR_OUT(), PARAM_INT_IN_REQ(), PARAM_DOUBLE_IN_REQ(),
+ * PARAM_STRING_IN_REQ(), PARAM_VECTOR_IN_REQ(), PARAM_INT_OUT_REQ(),
+ * PARAM_DOUBLE_OUT_REQ(), PARAM_VECTOR_OUT_REQ(), PARAM_STRING_OUT_REQ().
+ *
+ * Provide a link for a binding's "see also" documentation section, which is
+ * primarily (but not necessarily exclusively) used by the Markdown bindings
+ * This link can be specified by calling SEE_ALSO("description", "link"), where
+ * "description" is the description of the link and "link" may be one of the
+ * following:
+ *
+ * - A direct URL, starting with http:// or https://.
+ * - A page anchor for documentation, referencing another binding by its CMake
+ *      binding name, i.e. "#knn".
+ * - A link to a Doxygen page, using the mangled Doxygen name after a
+ *      '\@doxygen/', i.e., "@doxygen/mlpack1_1_adaboost1_1_AdaBoost".
+ */
+#ifdef __COUNTER__
+  #define BINDING_SEE_ALSO(DESCRIPTION, LINK) static \
+      mlpack::util::SeeAlso \
+      JOIN(io_programsee_also_dummy_object_, __COUNTER__) = \
+      mlpack::util::SeeAlso(DESCRIPTION, LINK);
+#else
+  #define BINDING_SEE_ALSO(DESCRIPTION, LINK) static \
+      mlpack::util::SeeAlso \
+      JOIN(JOIN(io_programsee_also_dummy_object_, __LINE__), opt) = \
+      mlpack::util::SeeAlso(DESCRIPTION, LINK);
+#endif
 /**
```

Comment:
```suggestion
#endif

/**
```
Another spacing fix. :)